ecs-cli/modules/cli/compose/project/project_parseV3.go

```diff
@@ -11,20 +17,93 @@ import (
 func (p *ecsProject) parseV3() (*[]containerconfig.ContainerConfig, error) {
 	logrus.Debug("Parsing v3 project...")
 
-	// TODO: parse v3, convert ServiceConfigs to ContainerConfigs
-	configDetails := types.ConfigDetails{}
+	// load and parse each compose file into a configFile
+	configFiles := []types.ConfigFile{}
+	for _, file := range p.ecsContext.ComposeFiles {
+
+		loadedFile, err := ioutil.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+		parsedFile, err := loader.ParseYAML(loadedFile)
+		if err != nil {
+			return nil, err
+		}
+		configFile := types.ConfigFile{
+			Filename: file,
+			Config:   parsedFile,
+		}
+		configFiles = append(configFiles, configFile)
+	}
+
+	// get build environment
+	env, err := getLocalEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	// get compose working dir
+	wrkDir, err := getWorkingDir(p.ecsContext.ComposeFiles[0])
+	if err != nil {
+		return nil, err
+	}
+
+	// build config details
+	configDetails := types.ConfigDetails{
+		WorkingDir:  wrkDir,
+		ConfigFiles: configFiles,
+		Environment: env,
+	}
+
+	// load config from config details
 	config, err := loader.Load(configDetails)
 	if err != nil {
 		return nil, err
 	}
 
+	// convert ServiceConfigs to ContainerConfigs
+	conConfigs := []containerconfig.ContainerConfig{}
 	for _, service := range config.Services {
-		convertDockerToContainerConfig(service)
+		cCon, err := convertDockerToContainerConfig(service)
+		if err != nil {
+			return nil, err
+		}
+		conConfigs = append(conConfigs, *cCon)
 	}
 
-	return &[]containerconfig.ContainerConfig{}, nil
+	// return container configs
+	return &conConfigs, nil
+}
+
+func convertDockerToContainerConfig(serviceConfig types.ServiceConfig) (*containerconfig.ContainerConfig, error) {
```

Comment:
Hm, what do you think about renaming this to simply `convertToContainerConfig`? Not sure I like "Docker" in the method name, and "ServiceConfig" seems too wordy. Given that this method is unexported, I'd say there's probably enough context that saying what you're converting to might be sufficient?

Reply:
agreed! will change